Xamarin.Forms.Core/ResourceDictionary.cs

```diff
@@ -36,6 +43,28 @@ public class ResourceDictionary : IResourceDictionary, IDictionary<string, objec
 			}
 		}
 
+		[TypeConverter(typeof(RDSourceTypeConverter))]
```

Comment:
ok, here's the magic...

if the type is attributed with a TypeConverter, the converter will be executed, and it's return value used as value for the setter.

during the execution of that converter we call SetAndLoadSource, which sets `_source` and loads the actual RD. When the converter returns the Uri, and the xaml engine sets it to this property, we no_op and exits.

this is the reason why it only works from xaml...